tests/functional/flows/test_validate_flow.py

```diff
@@ -0,0 +1,186 @@
+        assert "No scopes discovered" in result.output
+    else:
+        # Get the rows of the table
+        headers, table_data = _parse_table_content(result.output)[0]
+        # Check all fields are present
+        assert headers == ["Identity", "Scope"]
+        # Check the entries
+        assert len(table_data) == len(expected_table_data)
+        assert len(table_data[0]) == 2
+        assert table_data == expected_table_data
+
+
+def test_validate_flow_output_missing(run_line):
+    load_response_set("cli.flow_validate.missing")
+    result = run_line(["globus", "flows", "validate", "{}"])
+    assert "Discovered Scopes" not in result.output
+
+
+@pytest.mark.parametrize("input_schema", [None, {}])
+def test_validate_flow_input_schema(input_schema, run_line):
+    load_response_set("cli.flow_validate.none")
+    definition = {"StartAt": "a", "States": {"a": {"Type": "Pass", "End": True}}}
+
+    # Construct the command line
+    options = [(json.dumps(definition),)]
+    if input_schema is not None:
+        options.append(("--input-schema", json.dumps(input_schema)))
+
+    command = ["globus", "flows", "validate", *chain.from_iterable(options)]
+    result = run_line(command)
+
+    assert "No scopes discovered" in result.output
+
+    if input_schema is None:
+        # We should not have sent the input schema in the request
+        assert json.loads(get_last_request().body) == {"definition": definition}
+    else:
+        # We should have sent the input schema in the request
+        assert json.loads(get_last_request().body) == {
+            "definition": definition,
+            "input_schema": input_schema,
+        }
+
+
+def _parse_table_content(output):
+    """
+    Parse the output of a command, searching for tables in the output and returning
+    a list of headers and a list of rows (which are lists of cell values).
+
+    Expects a table with divider lines of the form `--- | --- | ---` and rows of the
+    form `value | value | value`.
+
+    Returns a list of tuples where each tuple represents a parsed table and is
+    comprised of a list of headers and a list of rows.
+
+    Raises a ValueError if no table is found in the output.
+    """
+    # Find the table divider
+    lines = output.split("\n")
+    divider_indices = [
+        i for i, line in enumerate(lines) if re.fullmatch(r"-+ \| [-| ]*", line)
+    ]
+
+    if not divider_indices:
+        raise ValueError("No table found in output")
+
+    tables = []
+    for divider_index in divider_indices:
+        # Parse the headers and rows
+        headers = [header.strip() for header in lines[divider_index - 1].split(" | ")]
+        rows = lines[divider_index + 1 :]
+        # Turn the rows into a table data as a list of lists
+        table_data = [
+            [cell.strip() for cell in row.split(" | ")] for row in rows if row
+        ]
+        tables.append((headers, table_data))
+
+    return tables
```

Comment:
I have nothing against having another way of parsing our own output on hand, for testing, but I think we have a couple of these kicking around already. It would be nice to do a cleanup sweep and produce something like a centralized fixture.

e.g.
```python
@pytest.fixture
def output_parser():
    return GlobusCLIOutputParser()


class GlobusCLIOutputParser:
    def parse_table(data: str) -> ParsedTable: ...
    def iter_all_tables(data: str) -> Iterable[ParsedTable]: ...
    def parse_record(data: str) -> ParsedRecord: ...
    def really_cool_stuff_here(data: str) -> YourNameOnThisType: ...
```

Reply:
Yeah, I didn't see one as I was writing. Agree—would prefer there are shared utilities for working with this.